lib/matplotlib/backends/backend_qt5agg.py

```diff
@@ -86,88 +84,37 @@ def paintEvent(self, e):
         if not hasattr(self, 'renderer'):
             return
 
-        if DEBUG:
-            print('FigureCanvasQtAgg.paintEvent: ', self,
-                  self.get_width_height())
-
-        if len(self.blitbox) == 0:
-            # matplotlib is in rgba byte order.  QImage wants to put the bytes
-            # into argb format and is in a 4 byte unsigned int.  Little endian
-            # system is LSB first and expects the bytes in reverse order
-            # (bgra).
-            if QtCore.QSysInfo.ByteOrder == QtCore.QSysInfo.LittleEndian:
-                stringBuffer = self.renderer._renderer.tostring_bgra()
-            else:
-                stringBuffer = self.renderer._renderer.tostring_argb()
-
-            refcnt = sys.getrefcount(stringBuffer)
-
-            # convert the Agg rendered image -> qImage
-            qImage = QtGui.QImage(stringBuffer, self.renderer.width,
-                                  self.renderer.height,
-                                  QtGui.QImage.Format_ARGB32)
-            if hasattr(qImage, 'setDevicePixelRatio'):
+        painter = QtGui.QPainter(self)
+
+        bbox_queue = self._bbox_queue or [
+            Bbox([[0, 0], [self.renderer.width, self.renderer.height]])]
+        self._bbox_queue = []
+        for bbox in bbox_queue:
+            l, b, r, t = map(int, bbox.extents)
+            w = r - l
+            h = t - b
+            reg = self.copy_from_bbox(bbox)
+            buf = reg.to_string_argb()
+            qimage = QtGui.QImage(buf, w, h, QtGui.QImage.Format_ARGB32)
+            if hasattr(qimage, 'setDevicePixelRatio'):
                 # Not available on Qt4 or some older Qt5.
-                qImage.setDevicePixelRatio(self._dpi_ratio)
-            # get the rectangle for the image
-            rect = qImage.rect()
-            p = QtGui.QPainter(self)
-            # reset the image area of the canvas to be the back-ground color
-            p.eraseRect(rect)
```

Comment:
This is important if users have alpha in the final rgba.  If this is not here text appears to 'get thicker' over time as the alpha stacks up.